sensibo/StateHandler.js

```diff
@@ -131,19 +129,27 @@ module.exports = (device, platform) => {
 				}
 
 				const sensiboNewACState = unified.sensiboFormattedACState(device, state)
-				const sensiboNewClimateReactState = unified.sensiboFormattedClimateReactState(device, state)
 
 				log.easyDebug(device.name, ' -> Setting New State:')
 				log.easyDebug(JSON.stringify(sensiboNewACState, null, 0))
 
-				if (platform.enableClimateReactAutoSetup) {
-					log.easyDebug(JSON.stringify(sensiboNewClimateReactState, null, 0))
-				}
-
 				try {
 					// send state command to Sensibo
 
 					if (platform.enableClimateReactAutoSetup) {
+						// NOTE: This below will NOT trigger the above setter, which is exactly what we want,
+						//       otherwise we could get into an infinite loop.
+						//
+						//       Essentially, if Climate React Auto Setup is enabled, Climate React will be
+						//       configured (and enabled) any time an AC state setting is changed.
+						//
+						//       That means that even if a user turned Climate React off via the appropriate
+						//       switch, the next AC state setting that is changed will re-enable it.
+
+						device.state.smartMode.enabled = true
```

Comment:
Yes, you're correct. I agree it's not intuitive. 

The problem is it's not clear to me what should be the outcome of turning climate react off if it was turned on as a result of the auto setup feature. Should it essentially disable the feature? What behavior would one expect here?

Also - say auto setup is on and the user turned off climate react then turned it back on, what is the expected behavior?

There's also the issue of what happens when the user turns climate react on/off directly from Sensibo's app/website. 

The current flow provides consistent behavior regardless of the origin of such commands.

When a user turns on auto setup I believe it kind of implies to the user that the "system" will set climate react automatically, that means it will override any manual settings that the user sets. If the user doesn't want that then he/she can turn auto setup off. 

I guess we could also have a switch for auto setup, per AC device, which would only be shown it if auto setup is enabled in the plugin's configuration.

Anyways, I'm open to suggestions re what to do here - not sure what is best.

Reply:
I think that makes a lot of sense, I will make the appropriate adjustments. I agree re configuration of the degrees" delta", it is on my todo list.

Will update the PR when I'm done.